lib/tasks/create_missing_whitehall_assets.rake

```
@@ -0,0 +1,53 @@
+    ["/government/uploads/system/uploads/attachment_data/file/211883/e0010125-response.pdf", "/government/uploads/system/uploads/attachment_data/file/224222/e0010125-response.pdf"],
+    ["/government/uploads/system/uploads/attachment_data/file/225280/SIN_Officer_Internal_Advert.docx", "/government/uploads/system/uploads/attachment_data/file/236779/Press_Reader_JD_and_Advert.pdf"],
+    ["/government/uploads/system/uploads/attachment_data/file/296338/mca-legislation-si.csv", "/government/uploads/system/uploads/attachment_data/file/297534/mca-legislation-si.csv"],
+    ["/government/uploads/system/uploads/attachment_data/file/490383/Local_Plans_Procedural_Guidance.pdf", "/government/uploads/system/uploads/attachment_data/file/531000/Procedural_Practice_in_the_Examination_of_Local_Plans_-_final.pdf"],
+    ["/government/uploads/system/uploads/attachment_data/file/627713/National_Lottery_Distribution_Fund_Investment_Account_2016-2017__web_.pdf", "/government/uploads/system/uploads/attachment_data/file/628399/National_Lottery_Distribution_Fund_Investment_Account_2016-2017__web_.pdf"]
+  ]
+
+  redirects.each do |(redirect_from, redirect_to)|
+    replacement = WhitehallAsset.find_by(legacy_url_path: redirect_to)
+
+    existing_asset = WhitehallAsset.where(legacy_url_path: redirect_from).first
+    if existing_asset
+      if existing_asset.replacement == replacement
+        puts "Asset identified by #{redirect_from} already exists so there's nothing to do"
+      else
+        puts "Asset identified by #{redirect_from} found but doesn't have matching redirect URL!"
+      end
+    else
+      puts "Asset identified by #{redirect_from} created"
+      WhitehallAsset.create!(
+        legacy_url_path: redirect_from,
+        file: File.open(placeholder_path),
+        replacement: replacement
```

I think it's possible (?) that `replacement` might be `nil`, in which case this `create!` call will raise an exception. Is that what you intended?

---

Yup. We want it to fail fast if something doesn't work as expected.